.github/workflows/preview-deploy.yml

```
@@ -0,0 +1,101 @@
+        run: |
+          # Generate unique preview token
+          PREVIEW_TOKEN=$(openssl rand -hex 16)
+          echo "PREVIEW_TOKEN=$PREVIEW_TOKEN" >> $GITHUB_ENV
+          
+          # Deploy with PR-specific environment
+          npx wrangler deploy \
+            --env pr-${{ github.event.number }} \
+            --var ENVIRONMENT:preview \
+            --var PREVIEW_TOKEN:$PREVIEW_TOKEN \
+            --config wrangler-preview.toml
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)